www/source/docs/blueprints.md

```diff
@@ -0,0 +1,58 @@
+# Blueprints
+Blueprints are default implementations of the build and runtime phases for your application. They also provide some default health check hooks where appropriate to ensure your application functioning reliably. Customized Blueprints can be created to facilitate reusability of common patterns in your organization for developing, building, and running your applications.
+
+## Getting Started
+### New Projects
+Simply run `hab plan init` to leverage blueprints. Habitat will look in your application code for known project types and the appropriate `pkg_blueprint`.
+
+If you would like to specify the desired blueprint to use, you can run `hab plan init java8-maven-tomcat`
+
+### Existing Plans
+
+To begin using Blueprints, you will need to add the appropriate `pkg_blueprints` 
+
+```bash
+pkg_name="my_java_app"
+....
+pkg_blueprints=("core/blueprints/java8-maven-tomcat")
+               ^     ^^^^^^^^^^^                    ^
+               |______|_______|_____________________|
+                 Arrays and additional namespace TBD -- just random thoughts at the moment.
+```
+* add `auto` package to `build_deps`
+* set toolchain variable
+
+#### Optional Variables
+* `pkg_src` - now accepts URI notation. Each blueprint contains a default URI which is relevant to the type of blueprint being used.
+*  `pkg_blueprint_opts` - Options which override some default runtime parameters. Eg: `--env DEV`. This operates similar to `pkg_svc_run` as the result of using this option will intelligently generate a hook file for you. However, you also can create the [hook file](#Run-Hooks) directly if you need to have more control over the resulting output.
+
+### Available Blueprints
+_The following list is a mock-up, and is does not currently reflect reality_:
+
+* `core/blueprint/ruby-mri1.9-rails-unicorn`
+* `core/blueprint/java8-maven-tomcat`
+* `core/blueprint/java8-maven-jboss`
+* `core/blueprint/java8-gradle-tomcat`
+* `core/blueprint/java7-ant`
+* `core/blueprint/python-unicorn`
+
+## Callbacks
+Each blueprint defines a set of callbacks which are unique to the blueprint type. This allows you to complex projects which may need to leverage different blueprints as needed within your project. Each of these callbacks define overrides of base level callbacks.
+
+### ruby _(This will likely live in the blueprint README.md)_
+#### do_bundler
+#### do_rake
+#### do_rack
+#### do_lotsmorestuff 
+
+## Run Hooks
+#### Health Checks
+Each blueprint comes with a default health check for your application.
+#### blueprint_opts
+Follow these directions if you wish to create the hook which would otherwise be created by the optional  `pkg_blueprint_opts` variable.
+
+*TODO: Write the directions*
+
+## Build Artifacts
+### Auto-Ingestion
+Each blueprint defines a default value for `pkg_src` in [URI syntax](https://tools.ietf.org/html/rfc3986). You can override this value within your plan should you application source live in a non-standard location.
```

Comment:
I'd like to see `pkg_source` ultimately become an optional variable, only to be used when your Plan isn't included with your codebase. At the moment we have no convenience variable for `$HAB_CACHE_SRC_PATH/$pkg_dirname` (which would be something like `/hab/cache/src/zlib-1.2.3` for most third-party core plans) but having such a variable (whether it's internal to the program or a public, user facing tunable) would help. Currently we have [`$PLAN_CONTEXT`](https://github.com/habitat-sh/habitat/blob/849fa27188b3c7cc3361d4edfd37c9a71e1d3bef/components/plan-build/bin/hab-plan-build.sh#L2414-L2440) which is used when we need to refer back to the `plan.sh` itself or the `./hooks` directory, etc.

We could change the build program slightly to the following:
1. if `pkg_source` is set, do what we do today and set an internal var `_src_root="$HAB_CACHE_SRC_PATH/$pkg_dirname"`
2. if `pkg_source` is not set, then set `_src_root="$(dirname "$PLAN_CONTEXT")"` (if we have the `./habitat/plan.sh` situation and `_src_root="$PLAN_CONTEXT"` if plan.sh is in the root of the codebase)

then each build phase wrapper uses `pushd "$_src_root"` (updating lines like [this](https://github.com/habitat-sh/habitat/blob/849fa27188b3c7cc3361d4edfd37c9a71e1d3bef/components/plan-build/bin/hab-plan-build.sh#L1858)).

Anyway, that's a sidetrack, but I'd really like to *not* use `$pkg_source` for plan.sh files that live with a codebase